test/StringUtils.test.cpp

```diff
@@ -1,8 +1,387 @@
+	EXPECT_EQ("1"s, NAS2D::stringFrom<long>(NAS2D::stringTo<long>("1"s)));
+
+	EXPECT_EQ("-1"s, NAS2D::stringFrom<long long>(NAS2D::stringTo<long long>("-1"s)));
+	EXPECT_EQ("0"s, NAS2D::stringFrom<long long>(NAS2D::stringTo<long long>("0"s)));
+	EXPECT_EQ("1"s, NAS2D::stringFrom<long long>(NAS2D::stringTo<long long>("1"s)));
+
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("-1"s)));
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("-1.0"s)));
+	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("-1.1"s)));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("0"s)));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("0.0"s)));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("1"s)));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("1.0"s)));
+	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<float>(NAS2D::stringTo<float>("1.1"s)));
+
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<double>(NAS2D::stringTo<double>("-1"s)));
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<double>(NAS2D::stringTo<double>("-1.0"s)));
+	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<double>(NAS2D::stringTo<double>("-1.1"s)));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<double> (NAS2D::stringTo<double>("0"s)));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<double> (NAS2D::stringTo<double>("0.0"s)));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<double> (NAS2D::stringTo<double>("1"s)));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<double> (NAS2D::stringTo<double>("1.0"s)));
+	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<double> (NAS2D::stringTo<double>("1.1"s)));
+
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("-1"s)));
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("-1.0"s)));
+	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("-1.1"s)));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("0"s)));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("0.0"s)));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("1"s)));
+	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<long double>(NAS2D::stringTo<long double>("1.1"s)));
+}
+
+
+TEST(String, stringFromToRoundtrip)
+{
+	using namespace std::literals::string_literals;
+	const std::string bigNum(40, '9');
+
+	EXPECT_EQ(""s, NAS2D::stringTo<std::string>(NAS2D::stringFrom<std::string>("")));
+	EXPECT_EQ("Hello World"s, NAS2D::stringTo<std::string>(NAS2D::stringFrom<std::string>("Hello World")));
+
+	EXPECT_EQ(true, NAS2D::stringTo<bool>(NAS2D::stringFrom<bool>(true)));
+	EXPECT_EQ(false, NAS2D::stringTo<bool>(NAS2D::stringFrom<bool>(false)));
+
+	EXPECT_EQ(65, NAS2D::stringTo<char>(NAS2D::stringFrom<char>(65)));
+	EXPECT_EQ(66, NAS2D::stringTo<char>(NAS2D::stringFrom<char>(66)));
```

Comment:
This could perhaps be written as:
```cpp
EXPECT_EQ('A', NAS2D::stringTo<char>(NAS2D::stringFrom<char>('A')));
EXPECT_EQ('B', NAS2D::stringTo<char>(NAS2D::stringFrom<char>('B')));
```